RiotShareExtension/Model/ShareExtensionManager.m

```diff
@@ -181,16 +189,23 @@ - (void)sendContentToRoom:(MXRoom *)room failureBlock:(void(^)())failureBlock
             }
             else if ([itemProvider hasItemConformingToTypeIdentifier:UTTypeImage])
             {
+                itemProvider.isLoaded = NO;
                 [itemProvider loadItemForTypeIdentifier:UTTypeImage options:nil completionHandler:^(NSData *imageData, NSError * _Null_unspecified error)
                  {
                      if (weakSelf)
                      {
-                         typeof(self) self = weakSelf;
-                         UIImage *image = [[UIImage alloc] initWithData:imageData];
-                         UIAlertController *compressionPrompt = [self compressionPromptForImage:image shareBlock:^{
-                             [self sendImage:imageData withProvider:itemProvider toRoom:room extensionItem:item failureBlock:failureBlock];
-                         }];
-                         [self.delegate shareExtensionManager:self showImageCompressionPrompt:compressionPrompt];
+                         itemProvider.isLoaded = YES;
+                         [self.pendingImages addObject:imageData];
+                         
+                         if ([self areAttachmentsFullyLoaded])
+                         {
+                             UIImage *firstImage = [UIImage imageWithData:self.pendingImages.firstObject];
+                             UIAlertController *compressionPrompt = [self compressionPromptForImage:firstImage shareBlock:^{
+                                 [self sendImages:self.pendingImages withProvider:itemProvider toRoom:room extensionItem:item failureBlock:failureBlock];
```

Comment:
I don't understand here why only one itemProvider is used for all the pending images. Is that normal?